src/main/scala/analysis/MemoryModelMap.scala

```diff
@@ -21,23 +23,23 @@ class MemoryModelMap {
   private val allStacks = mutable.Map[String, List[StackRegion]]()
 
   // Add a range and object to the mapping
-  def add(offset: BigInt, regionType: MemoryRegion): Unit = {
-    regionType match {
+  def add(offset: BigInt, region: MemoryRegion): Unit = {
+    region match {
       case s: StackRegion =>
         val stackMap = rangeMap.stackMap
         if (stackMap.isEmpty) {
-          stackMap(RangeKey(offset, MAX_BIGINT)) = regionType
+          stackMap(RangeKey(offset, MAX_BIGINT)) = region.asInstanceOf[StackRegion]
         } else {
           stackMap.keys.maxBy(_.end).end = offset - 1
-          stackMap(RangeKey(offset, MAX_BIGINT)) = regionType
+          stackMap(RangeKey(offset, MAX_BIGINT)) = region.asInstanceOf[StackRegion]
         }
       case d: DataRegion =>
         val dataMap = rangeMap.dataMap
         if (dataMap.isEmpty) {
-          dataMap(RangeKey(offset, MAX_BIGINT)) = regionType
+          dataMap(RangeKey(offset, MAX_BIGINT)) = region.asInstanceOf[DataRegion]
         } else {
           dataMap.keys.maxBy(_.end).end = offset - 1
-          dataMap(RangeKey(offset, MAX_BIGINT)) = regionType
+          dataMap(RangeKey(offset, MAX_BIGINT)) = region.asInstanceOf[DataRegion]
```

Comment:
You can just use `s` and `d` instead of casting region

Reply:
Resolved. [Upcoming merge from yousif-memory-region-analysis]